### PR TITLE
[RW-6692][risk=low] Eliminate spurious changes to user.lastModifiedTime

### PR DESCRIPTION
**Pull request by calbach**
- Leverage `@UpdateTimestamp` for automatic timestamp updates via Hibernate, which resolves an issue with excessive updates to `lastModifiedTime` (i.e. on every cron run).
- Switch to `@CreationTimestamp` for consistency, this way these columns have an identical value on initial insertion

## Why this works
These annotations only have an effect if you make a change which would modify the database row. A no-op `save()` does not change the lastModified time, which is exactly the desired behavior for offline cronjobs etc. This should reduce RDR export user selection to expected levels, since it looks at `lastModifiedTime` to determine whether to send data.

Tested by locally triggering crons, profile updates, user create and observing the expected `last_modified_time` in the database.

Filed https://precisionmedicineinitiative.atlassian.net/browse/RW-6743 to explore this for the rest of the models.

One downside: this approach is less testable than the clock approach. Hibernate does not make it easy to stub out this value for testing purposes: https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/tuple/TimestampGenerators.java

### CODE REVIEW

File: api/src/test/java/org/pmiops/workbench/billing/FreeTierBillingServiceTest.java [hunk 1]
```diff
@@ -338,29 +330,16 @@ public void checkFreeTierBillingUsage_workspaceMissingCreatorNoNPE() {
...
-
-    // we update the user and should see this last modified time
-    final Instant time2 = START_INSTANT.plusSeconds(1000);
-    CLOCK.setInstant(time2);
 
     assertThat(freeTierBillingService.maybeSetDollarLimitOverride(user, 200.0)).isTrue();
     verify(mockUserServiceAuditor)
         .fireSetFreeTierDollarLimitOverride(user.getUserId(), null, 200.0);
     assertWithinBillingTolerance(freeTierBillingService.getUserFreeTierDollarLimit(user), 200.0);
-    assertThat(userDao.findUserByUserId(user.getUserId()).getLastModifiedTime())
```
Comment (calbach) [inline api/src/test/java/org/pmiops/workbench/billing/FreeTierBillingServiceTest.java:351]: FYI @jmthibault79 my interpretation is that these were not critical assertions to the test. Let me know if you think this was indirectly trying to detect something else which I should now cover differently.

Comment (jmthibault79) [reply api/src/test/java/org/pmiops/workbench/billing/FreeTierBillingServiceTest.java:351]: I think the logic here was: we set the lastModified in the service, so let's confirm that we did so.

So it's correct to remove these.

**Review by petesantos — APPROVED** [review]
LGTM
Is there any check we would want to do to ensure the timestamp is not changing?

**Comment by calbach** [comment]
@petesantos good call, I added a unit test for this scenario to enforce this behavior going forward.

**Pull request merged**